### DISCUSSION TITLE
Issues with API - specifically the URL

### DISCUSSION THREAD

**Discussion by Normanras** [blakeblackshear/frigate#1556] [General]
Up until a few days ago, Frigate had been working really, really well. The particular way I was using it - and now issue - is via Telegram commands and notifications. In particular I a few commands that would send me the last or best picture. The URL being taken straight from documentation:
`http://ip-address:5000/api/location/object/best.jpg` or `http://ip-address:5000/api/driveway/latest.jpg`.

Not sure what happened, but the commands still send me a picture, but most of them are just black. If I go into the media library in Home Assistant, the pictures exist and I see the one that _should_ have been sent. In the Frigate UI, the small icon for each object (that previews the image) is black, but when I open up the list of events and pictures, I see all the pictures there. The same pictures as in the media library. 

I don't think (yet) that this is a bug, but perhaps something I've done and I wanted to see if anyone else has experienced this. The only changes that have happened in the last few days that I think could have caused this:

- HA updates to 8.6, 8.7, and then tonight 8.8
- I renamed the entity ids of a few automation that use either the commands or send me pictures after a MQTT message (which also doesn't work now). I shortened the entities to `automation.a_really_long_id` to `automation.location_object`

I don't believe I touched my YAML file and just looked over it before posting... it seems good. If anyone has any thoughts, please let me know!

**Comment by ozett** [comment]
i have "best.jpg", but not "last.jpg"

![image](https://user-images.githubusercontent.com/5937811/130023366-9f07e2cf-95b5-4d23-aa4a-efffd788045d.png)

**Reply by Normanras** [reply]
Apologies - it should say "latest". I will update my post. Also, the latest URL would be without the object. It will just pull a current snap shot from that camera, even if nothing has triggered it.

**Comment by blakeblackshear** [comment]
`latest.jpg` should only be black briefly on startup before the first frame has made it to frigate. `best.jpg` is expected to be black until the next object is detected after a restart.

Is this happening consistently? I have seen this happen occasionally, but it has been rare. Anything in the logs? I don't believe this has anything to do with the HA updates.

Of note, I think most people have moved over to using the `frigate/events` mqtt topic and fetching snapshots/thumbnails using the specific event id. The `best.jpg` endpoint is really leftover from a time before objects were tracked.

**Reply by Normanras** [reply]
Wow, yes lots of errors. I can't believe I didn't check these sooner. They seem to repeat themselves in this block:

`frigate.video                  INFO    : porch: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
frigate.video                  INFO    : porch: ffmpeg process is not running. exiting capture thread...
ffmpeg.porch.detect            ERROR   : [flv @ 0x556ee254a0] Timestamps are unset in a packet for stream 0. This is deprecated and will stop working in the future. Fix your code to set the timestamps properly
ffmpeg.porch.detect            ERROR   : av_interleaved_write_frame(): Connection reset by peer
ffmpeg.porch.detect            ERROR   : Finishing stream 1:0 without any data written to it.
ffmpeg.porch.detect            ERROR   : [flv @ 0x556ee254a0] Failed to update header with correct duration.
ffmpeg.porch.detect            ERROR   : [flv @ 0x556ee254a0] Failed to update header with correct filesize.
ffmpeg.porch.detect            ERROR   : Error writing trailer of rtmp://127.0.0.1/live/porch: Connection reset by peer`

I don't think any of this info is sensitive, but you would know best @blakeblackshear. I'll edit out if you see anything that shouldn't be public. 

I don't have any ffmpeg rules in my yaml except the basic of declaring ffmpeg - no extra parameters. I think I'll have to pare down my frigate file and see if anything changes. Start with the basics.

**Reply by ozett** [reply]
i had mostly to check and adjust height and width of stream and in the config for similar error-msg..
after that i checked the ffmpeg params... good luck

**Reply by blakeblackshear** [reply]
I don't know what type of camera you have, but these errors are expected to happen occasionally unless you have very stable PoE cameras. Normally, frigate will restart ffmpeg and recover.

**Reply by Normanras** [reply]
Thanks all - yeah height and width and all other parameters have been functioning flawless (you're truly amazing, blake!) other than this random glitch. I would say I have stable PoE cameras, but I can't be 100% sure. I've already rebuilt the add-on, but let me change the yaml and restart the add-on. Hopefully, that will fix it.
